### DISCUSSION TITLE
theme.ts

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#5208] [Q&A]
How do I get the colorMode value inside the colors, if I can't use useColorMode() ?

![s](https://user-images.githubusercontent.com/68085898/145305742-ececeb43-cbbf-4eb4-a7da-d8834584e2e7.png)

**Comment by primos63** [comment]
You can refer to a component within the default theme like [Button](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/button.ts). Easiest way to get there is going through the docs and clicking on the `View theme source` button at the top of the page.

You'll see it imports `mode` from `@chakra-ui/theme-tools`
`mode(light, dark)(props)`
`mode` is a function that takes the light mode and dark mode setting and returns a function with takes `props` as an argument. It uses `props` to get the `colorMode` and returns your light or dark mode setting based on it's value. 

Colors isn't a component though so you cannot use a function there.

You can do this:
```javascript
const theme = extendTheme({
  colors: {
    primary: {
      light: "#f7e",
      dark: "#fff"
    }
  },
  components: {
    Button: {
      variants: {
        solid: (props) => {
          return {
            color: mode("primary.light", "primary.dark")(props)
          };
        }
      }
    }
  }
}
```  
In your app
```jsx
<Button colorScheme='blue'>Button</Button>
```
would produce this in light mode
![image](https://user-images.githubusercontent.com/2768849/145310207-72184b71-0639-4c46-a0c8-4ee3cca2225e.png)
and this in dark mode
![image](https://user-images.githubusercontent.com/2768849/145310261-7ed84504-f4f0-4cf3-a6e2-ac80c7e534ca.png)

**Reply by ghost** [reply]
I'm using typescript what is the props type? please

**Reply by primos63** [reply]
If you look at a component module, you'll see that props is of type [StyleFunctionProps](https://github.com/chakra-ui/chakra-ui/blob/6fccb264cd0486685cd82bab5f209deab52ac3f1/packages/theme-tools/src/component.ts#L37)
